### PR TITLE
compiler: enable ORC when running the leak test

### PR DESCRIPTION
**Pull request by zerbina**
## Summary

Even when ORC is disabled at run-time, cells potentially part of
reference cycles are added to potential-cycle-root list, preventing
the leak sanitizer from detecting cycles not being free.

The `leakTest` define is now passed to the compiler when leak-
testing, with ORC enabled when the define is present.

### CODE REVIEW

**Comment by alaviss** [comment]
How about we set it to run at exit? Because even now a single bootstrap cycle takes 70s for leak tests, so I'm not sure if we want to have it running for the entirety of compilation.

Though, being merge CI means time doesn't matter as much.

**Review by alaviss — APPROVED** [review]
Actually I just looked at the numbers again and they looks fairly normal, so I don't think there's much to do here.

**Comment by zerbina** [comment]
> How about we set it to run at exit?

Sure, that works too.

> Because even now a single bootstrap cycle takes 70s for leak tests

Hm, do you mean locally for you? In the CI, the (NimSkull part) of the second and third iteration only take ~27 seconds each (the number comes from [here](https://github.com/nim-works/nimskull/actions/runs/7596243536/job/20689735177)).

**Review by saem — APPROVED** [review]
tl;dr: I think we should merge as is

My understanding is that if we were to add it at the end we'd put a `GC_enableOrc()` at the end of `mainCommand` and also force a collection, in the presence of the `leakTest` define.

If it's that tiny, then that's tempting for faster CI. Although, and my memory could be playing tricks on me, but I don't believe we've got leak detection testing in our regular tests. So maybe we're better off deferring the optimization until we've got more core tests in place; I could just be paranoid.

**Comment by alaviss** [comment]
> In the CI, the (NimSkull part) of the second and third iteration only take ~27 seconds each (the number comes from [here](https://github.com/nim-works/nimskull/actions/runs/7596243536/job/20689735177)).

Yea that's why I revised later that the numbers looked alright.

I must have imagined the 70s figure because I couldn't find it anywhere now. Though a forced-collection might be useful since I don't think we perform full collect at the end of the program by default.

**Comment by alaviss** [comment]
/merge

Since we have 2 approvals already, let's get this in.

**Comment by alaviss** [comment]
/merge

Just some GitHub issues, should be fine now with another go.
